pyproject.toml

```diff
@@ -56,35 +56,35 @@ docs-gen = [
     "mkdocstrings[python] >= 0.19.0, < 0.20.0",
 ]
 docs-lint = [
-    "pydocstyle",
-    "darglint",
-    "tomli",  # Needed by pydocstyle to read pyproject.toml
+    "pydocstyle >= 6.3.0, < 7",
+    "darglint >= 1.8.1, < 2",
+    "tomli >= 2.0.1, < 3",  # Needed by pydocstyle to read pyproject.toml
 ]
 format = [
-    "black",
-    "isort",
+    "black >= 23.1.0, < 24",
+    "isort >= 5.12.0, < 6",
 ]
 nox = [
-    "nox",
-    "toml",
+    "nox == 2022.11.21",
+    "toml >= 0.10.2, < 1",
 ]
 pytest = [
-    "pytest",
-    "pytest-cov",
-    "pytest-mock",
-    "pytest-asyncio",
-    "time-machine",
-    "async-solipsism",
+    "pytest >= 7.2.1, < 8",
+    "pytest-cov >= 4.0.0, < 5",
+    "pytest-mock >= 3.10.0, < 4",
+    "pytest-asyncio >= 0.20.3, < 1",
+    "time-machine >= 2.9.0, < 3",
+    "async-solipsism >= 0.5, < 1",
 ]
 mypy = [
-    "mypy",
-    "pandas-stubs",
-    "grpc-stubs",
+    "mypy >= 1.0.1, < 2",
+    "pandas-stubs == 1.5.3.230304",
+    "grpc-stubs == 1.24.12",
```

Comment:
Why are these still fixed to one version? Did they introduce breaking changes in a patch release in the past?

Reply:
(if so, please add a comment, so people don't think it was an overlook)

Reply:
I'll add a comment

Reply:
What about `pandas-stubs`?

Reply:
added the comment for `grpc-stubs`. Do you want to remove the strict pin for `pandas-stubs`?

Reply:
For me it looks semver-ish enough to be less strict and change it if it breaks or if we do a little reach and find out they break it often (as a general approach). But I guess ela is rights now that the old packages are gone we don't even need pandas anymore, so we can just remove it (this PR or another one, up to you).

Reply:
I'll push now a PR to remove pandas(-stubs)

Reply:
https://github.com/frequenz-floss/frequenz-sdk-python/pull/261